honeycombio/honeycomb-opentelemetry-java
# maint: Add dependency license files

## Which problem is this PR solving?
It is a legal requirement to include the license of any software used to build distributable artifacts. This PR adds the [Gradle-License-Report](https://github.com/jk1/Gradle-License-Report) plugin to the agent's Gradle file so that we have a job to run to generate license files for dependencies. This is also how [upstream Opentelemetry Java handles dependency licenses](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/dc2c4f630a95a4fc8f04d39a374ac297736c99b9/javaagent/build.gradle.kts#L6).

It also updates the releasing.md to include the steps to ensure this is updated as part of the release process.

## Short description of the changes
- Updates the agent's gradle file to include the license plugin
- Adds all licenses and a license report under the `/licenses` directory
- Updates the release notes to check whether licences have been included / updated for all dependencies.

## How to test
1. Pull down this branch
2. Delete the `/licenses` directory
3. Run `./gradlew generateLicenseReport`
4. You should see a `/licenses` folder with licenses and a markdown report file
